IBM/ibm-cert-manager-operator
# Post install delegation checker

Implements a new controller `PostDelegationChecker` to check whether delegation to another cert-manager is needed after this operator has already been installed.

Situation that this covers is when this operator is already installed and the operands are already created. Then user installs another cert-manager onto the cluster.

## How to test

1. See how to test with by running bundle https://github.com/IBM/ibm-cert-manager-operator/blob/release-ltsr/CONTRIBUTING.md#testing-bundle-with-olm
2. After running the bundle, operator pod should be running on the cluster
3. Create the operands by running `oc apply -f config/samples/operator_v1_certmanagerconfig.yaml`
4. Verify that operands are created
5. Install another cert-manager, e.g. Red Hat cert-manager
6. Verify that IBM operands are deleted, and RH cert-manager is running
7. Verify that you can create a self-signed Issuer and CA certificate
